### ISSUE TITLE
Timepicker component

### ISSUE THREAD

**Issue by inwardmovement** [adobe/react-spectrum#5584]
### Provide a general summary of the feature here

It would be awesome to provide a TimePicker component, to complement the TimeField component.

### 🤔 Expected Behavior?

The TimePicker component would provide the structure and accessibility to build a dial/clock/wheel input, like [Material Design TimePicker](https://m3.material.io/components/time-pickers/).

![TimePicker image](https://lh3.googleusercontent.com/Bu5dxRtOEDkj5X7bDpk695Ju_jidUnSNfmdPncONir3-VJz_UlSVccSHcFNKc4TkRI3pGyWIfeLeh5Z6ux8hH1_t2HYNZvMRmHFrfRaYdgh3FA=s0)

### 😯 Current Behavior

N/A

### 🔦 Context

N/A

**Comment by reidbarber** [comment]
I'm not sure this is something we'll be implementing or maintaining, given that is doesn't exist in Adobe's Spectrum design system currently. I think someone could potentially build this with some inspiration from the [useSlider](https://react-spectrum.adobe.com/react-aria/useSlider.html) implementation though.

**Comment by zaru** [comment]
I think having a list-style UI for selecting hours and minutes, similar to the browser’s default <input type="time" />, would be convenient.

<img width="269" alt="Image" src="https://github.com/user-attachments/assets/8d2a9583-2eee-4040-851d-466a01c12437" />

**Comment by devongovett** [comment]
I don't think we have plans to implement anything like this in the near future. We already have [TimeField](https://react-spectrum.adobe.com/react-aria/TimeField.html). Entering a time by typing it with a keyboard seems more efficient to me anyway.
